node/core/pvf/src/worker_common.rs

```diff
@@ -357,25 +363,35 @@ impl futures::Future for WorkerHandle {
 
 	fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
 		let me = self.project();
-		match futures::ready!(AsyncRead::poll_read(me.stdout, cx, &mut *me.drop_box)) {
-			Ok(0) => {
-				// 0 means `EOF` means the child was terminated. Resolve.
-				Poll::Ready(())
-			},
-			Ok(_bytes_read) => {
-				// weird, we've read something. Pretend that never happened and reschedule ourselves.
-				cx.waker().wake_by_ref();
-				Poll::Pending
+		// Create a `ReadBuf` here instead of storing it in `WorkerHandle` to avoid a lifetime
+		// parameter on `WorkerHandle`. Creating the `ReadBuf` is fairly cheap.
+		let mut read_buf = ReadBuf::new(&mut *me.drop_box);
```

Comment:
We're forced to pass `ReadBuf`, which takes a lifetime parameter, to `poll_read`. I figured it should be fine to instantiate it on each `poll` since it's an inline function, but if we want to store it in the struct instead, we may want to use e.g. `rental`.